docs/NEW_NEAR_CLI_INTERFACE.md

```diff
@@ -0,0 +1,191 @@
+- All direct children commands of a single parent command should be aligned (either represent an action or a resource, but never a mix of those on the same hierarchy level): `contract` -> `state` (resource) and `contract` -> `deploy` (action) are not aligned, so it should be either `contract` -> `get-state` + `contract` -> `deploy` or `contract` -> `state` -> `view` + `contract` -> `code` -> `deploy`
+- Interactive mode should look like: `command - description`. It will help people to learn the commands.
+
+## `Core NEAR CLI` commands
+
+```
+account
+  - view-account <account-id> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+ 
+  - create-sub-account <new-account-id> <initial-balance> 
+    (we will treat everything after the first dot in account id as the parent account (transaction signer))
+    - with-plaintext-public-key "ed25519:..." network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+    - with-generated-keypair network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+
+  - delete-account <account-id> beneficiary <beneficiary-account-id> network <"mainnet"|"testnet"|...>
+    - transaction signature options here (see below)
+
+  - list-keys <account-id> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+
+  - add-key <account-id>
+    - use-plaintext-public-key "ed25519:..." network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+    - generate-keypair network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+
+  - delete-key <account-id> <public-key> network <"mainnet"|"testnet"|...>
+    - transaction signature options here (see below)
+
+  - propose-stake <account-id> <stake-amount-in-NEAR> <validation-node-public-key>
+    (maybe we should extract it into the validators extension)
+    - transaction signature options here (see below)
+
+  - TODO: "import account" (aka "login") commands
+```
+
+```
+contract
+  - call-view-function <account-id> <function-name> <function-args> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+
+  - call-change-function <account-id> <function-name> <function-args> --prepaid-gas <prepaid-gas> --attached-deposit <deposit-amount> network <"mainnet"|"testnet"|...>
+    - transaction signature options here (see below)
+  
+  - deploy <account-id> use-file <path-to-wasm-file>
+
+    - with-init-call <function-name> <function-args> --prepaid-gas <prepaid-gas> --attached-deposit <deposit-amount>
+      - transaction signature options here (see below)
+    - no-init-call
+      - transaction signature options here (see below)
+
+  - download <account-id> to-folder <path-to-download-folder> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
```

Comment:
perhaps we can make this more clear and say `download-wasm` or something because it's not immediately clear what you're downloading